test/Parse/multiline_string.swift

```diff
@@ -109,6 +109,28 @@ _ = """
 // CHECK: "Twelve\nNu"
 
 _ = """
+  newline \
+  elided
+  """
+
+// CHECK: "newline elided"
+
+let b = """
+  newline \
+  \("""
+    substring1 \
+    \("""
+      substring2 \
+      substring3
+      """)\
+    """) \
+  elided
+  """
+
+// CHECK: "substring1 "
+// CHECK: "substring2 substring3"
+
+_ = """
```

Comment:
Not sure if this is tested here, but is all whitespace after the \ skipped?

Also, we should explicitly test behavior when non-whitespace follows the backslash, including comments, as an error.